ddl/ddl_worker_test.go

```diff
@@ -481,6 +487,8 @@ func (s *testDDLSuite) TestCancelJob(c *C) {
 			return
 		}
 	}
+	tc.onJobUpdated = hookCancelFunc
+	tc.onJobRunBefore = hookCancelFunc
```

Comment:
Why change this code?

Reply:
we need to hook before the job running.

Reply:
I know the function of `onJobRunBefore`, but why we need to add this hook to it?

Reply:
If not, you can not hook the job when the state is None. when the job updated, the job.State is definitely not None.